lib/sqlalchemy/dialects/mysql/types.py

```diff
@@ -422,17 +422,22 @@ def result_processor(self, dialect, coltype):
         time = datetime.time
 
         def process(value):
-            # convert from a timedelta value
             if value is not None:
-                microseconds = value.microseconds
```

Comment:
the types.py logic here works for all of the other MySQL dialects, even mysql connector and the defunct OurSQL.  so that would make pyodbc the exception case.  So this change should be in mysql/pyodbc.py, not in types.py.   in mysql/pyodbc.py add a new _pyodbcTIME class which subclasses this one, and then apply the alternate result_processor.  Then add this _pyodbcTIME class to the pyodbc-level "colspecs" dictionary, which itself also needs to be added:

```
# lib/sqlalchemy/dialects/mysql/pyodbc.py

class MySQLDialect_pyodbc(...):
     colspecs = util.update_copy(MySQLDialect.colspecs, {TIME: _pyodbcTIME})
```